gossip/src/crds_value.rs

```diff
@@ -95,30 +95,31 @@ impl CrdsValueLabel {
 }
 
 impl CrdsValue {
-    pub fn new_unsigned(data: CrdsData) -> Self {
+    pub fn new(data: CrdsData, keypair: &Keypair) -> Self {
+        let bincode_serialized_data = bincode::serialize(&data).unwrap();
+        let signature = keypair.sign_message(&bincode_serialized_data);
+        Self { signature, data }
+    }
+
+    #[cfg(test)]
+    pub(crate) fn new_unsigned(data: CrdsData) -> Self {
```

Comment:
is there any reason to keep `new_unsigned()`? I know it's now `pub(crate)` and only used for tests. but wonder if it's worth just getting rid of completely so someone doesn't accidently use `new_unsigned()`. plus, future tests could possibly miss issues if `new_unsigned()`

Reply:
That is true and I wouldn't have added it in the first place.
But it is legacy code and used so many places.

I guess the motivation has been not to waste time signing crds-values for tests.

Reply:
ok ya i'm fine leaving it for now. I get the reasoning. but it's probably worth removing in the future